Makefile

```diff
@@ -951,7 +951,111 @@ ifeq "$(OPENMP)" "true"
 endif
 
 
-pio_test:
+openacc_test : 
+ifeq "$(OPENACC)" "true"
+	@#
+	@# First ensure that both FFLAGS_ACC and CFLAGS_ACC are not blank
+	@# If these are not set for a target, then OpenACC most likely cannot compile
+	@#
+	@echo "Checking if FFLAGS_ACC and CFLAGS_ACC are defined for [$(BUILD_TARGET)]..."
+	@if ([ -z "$(FFLAGS_ACC)" ] && [ -z "$(CFLAGS_ACC)" ]); then \
+	   (echo "*********************************************************"; \
+	    echo "ERROR: OPENACC=true was specified, but [$(BUILD_TARGET)] build target does not seem to support OpenACC:"; \
+	    echo "    FFLAGS_ACC and CFLAGS_ACC are both undefined for [$(BUILD_TARGET)] in the top-level Makefile."; \
+	    echo "Please set these variables to appropriate OpenACC compilation flags in the [$(BUILD_TARGET)] target to enable OpenACC support."; \
+	    echo "*********************************************************"; exit 1)\
+	else \
+	   echo "=> FFLAGS_ACC or CFLAGS_ACC defined"; \
+	fi
+
+	@#
+	@# Create test C and FORTRAN programs that look for OpenACC header file and parallelize a loop
+	@#
+	@printf "#include <openacc.h>\n \
+	         &int main(){\n \
+	         &    int n_devs=acc_get_num_devices( acc_device_current );\n \
+	         &    int i,n=0;\n \
+	         &    #pragma acc kernels\n \
+	         &    for (i=0; i<10; i++)\n \
+	         &      n=n+i;\n \
+	         &    return 0;\n \
+	         &}\n" | sed 's/^[ \t]*&//' > openacc.c
+	@printf "program openacc\n \
+		&    use openacc\n \
+		&    integer :: i,n=0,n_devs=0\n \
+		&    n_devs=acc_get_num_devices( acc_device_current )\n \
+		&    !\$$acc kernels\n \
+		&    do i=0,10\n \
+		&      n=n+i\n \
+		&    end do\n \
+		&    !\$$acc end kernels\n \
+		&end program\n" | sed 's/^[ \t]*&//' > openacc.f90
```

Comment:
For the C test program, indentation uses a tab followed by spaces, while for the Fortran test program, it's two tabs. It might be nice to make these consistent (and my preference would lean towards the former).

Reply:
Thanks for catching that. I agree that the test programs should be consistent and also should be indented by one tab and then spaces.